tests/setup_test.go

```diff
@@ -1029,27 +1030,89 @@ func TestCreateProject(t *testing.T) {
 func TestUpdateProject(t *testing.T) {
 	t.Parallel()
 
-	tests := []struct {
-		name               string
-		passVarsToForkedPR bool
-		expectedPre        bool
-		expectedPost       bool
-	}{
-		{
-			name:               "test project update with pass-vars-to-forked-pr true",
-			passVarsToForkedPR: true,
-			expectedPre:        false,
-			expectedPost:       true,
-		},
-		{
-			name:               "test project update with pass-vars-to-forked-pr false",
-			passVarsToForkedPR: false,
-			expectedPre:        false,
-			expectedPost:       false,
-		},
-	}
+	t.Run("test update PassVarsToForkedPR in users's project", func(t *testing.T) {
+		dir := t.TempDir()
 
-	for _, tt := range tests {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sc := setup(ctx, t, dir, withGitea(true))
+		defer sc.stop()
+
+		giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+		giteaToken, token := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+
+		giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, token)
+
+		_, project := createProject(ctx, t, giteaClient, gwClient)
+		if project.PassVarsToForkedPR != false {
+			t.Fatalf("expected PassVarsToForkedPR false, got %v (pre-update)", project.PassVarsToForkedPR)
+		}
+
+		project = updateProject(ctx, t, giteaClient, gwClient, project.ID, false)
+		if project.PassVarsToForkedPR != false {
+			t.Fatalf("expected PassVarsToForkedPR false, got %v (pre-update)", project.PassVarsToForkedPR)
+		}
+
+		project = updateProject(ctx, t, giteaClient, gwClient, project.ID, true)
```

Comment:
remove `updateProject` function and directly call the api here as done in other tests.